### PR TITLE
Add cluster configuration page

### PR DESCRIPTION
**Pull request by pipo02mix**
### What this PR does / why we need it
Towards https://gigantic.slack.com/archives/CRZN9GLJW/p1723797774944489?thread_ts=1723737574.001869&cid=CRZN9GLJW

### Things to check/remember before submitting

- If you made content changes

    - Run `make lint dev` to render and proofread content changes locally.
    - Bump `last_review_date` in the front matter header if you reviewed the entire page.

### CODE REVIEW

**Comment by Gacko** [comment]
I wanted to review, but review dog already did the job. Just kidding, could you consider the suggestions? I'll come back later. :)

**Comment by pipo02mix** [comment]
@Gacko yes please!

File: src/content/overview/fleet-management/cluster-management/cluster-concepts/cluster-configuration/_index.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+user_questions:
+  - What is cluster configuration in the Giant Swarm platform?
+  - How can I customize the configuration of a cluster in the Giant Swarm platform?
+---
+
+The cluster configuration is managed thanks to the [app platform]({{< relref "/vintage/getting-started/app-platform/app-configuration" >}}) configuration options.
+
+In the cluster, there are 3 main configuration sources:
+
+- Default provider-independent app config (comes from cluster chart; if some app has this, then this should be probably moved directly to the app repo itself)
```
Comment (pipo02mix) [inline src/content/overview/fleet-management/cluster-management/cluster-concepts/cluster-configuration/_index.md:21]:
```suggestion
- Default provider-independent app configuration (comes from cluster chart; if some app has this, then this should probably be moved directly to the app repository itself)
```

File: src/content/overview/fleet-management/cluster-management/cluster-concepts/cluster-configuration/_index.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+- Provider-specific app config (comes from cluster-<provider>)
```
Comment (pipo02mix) [inline src/content/overview/fleet-management/cluster-management/cluster-concepts/cluster-configuration/_index.md:22]:
```suggestion
- Provider-specific app configuration (comes from cluster-<provider>)
```

File: src/content/overview/fleet-management/cluster-management/cluster-concepts/cluster-configuration/_index.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+- Customer-specified app config (come from cluster-<provider> Helm values, it is probably specified in customer's gitops repo).
```
Comment (pipo02mix) [inline src/content/overview/fleet-management/cluster-management/cluster-concepts/cluster-configuration/_index.md:23]:
```suggestion
- Customer-specified app configuration (come from cluster-<provider> Helm values, it's probably specified in customer's GitOps repository).
```

**Review by Gacko — APPROVED** [review]

**Pull request merged**